vali_objects/utils/p2p_syncer.py

```diff
@@ -0,0 +1,114 @@
+    async def send_checkpoint(self):
+        """
+        serializes checkpoint json and transmits to all validators via synapse
+        """
+        # get our current checkpoint
+        now_ms = TimeUtil.now_in_millis()
+        checkpoint_dict = generate_request_core(time_now=now_ms)
+
+        # serialize the position data
+        positions = checkpoint_dict['positions']
+        # 24 hours in milliseconds
+        max_allowed_t_ms = TimeUtil.now_in_millis() - AUTO_SYNC_ORDER_LAG_MS
+        for hotkey, positions in positions.items():
+            new_positions = []
+            positions_deserialized = [Position(**json_positions_dict) for json_positions_dict in positions['positions']]
+            for position in positions_deserialized:
+                new_orders = []
+                for order in position.orders:
+                    if order.processed_ms < max_allowed_t_ms:
+                        new_orders.append(order)
+                if len(new_orders):
+                    position.orders = new_orders
+                    position.rebuild_position_with_updated_orders()
+                    new_positions.append(position)
+                else:
+                    # if no orders are left, remove the position
+                    pass
+
+            positions_serialized = [json.loads(str(p), cls=GeneralizedJSONDecoder) for p in new_positions]
+            positions['positions'] = positions_serialized
+
+        # compress json and encode as base64 to keep as a string
+        checkpoint_str = json.dumps(checkpoint_dict, cls=CustomEncoder)
+        compressed = gzip.compress(checkpoint_str.encode("utf-8"))
+        encoded_checkpoint = base64.b64encode(compressed).decode("utf-8")
+
+        # get axons to send checkpoints to
+        dendrite = bt.dendrite(wallet=self.wallet)
+        validator_axons = self.metagraph.axons
+
+        if len(validator_axons) == 0:
+            bt.logging.info(f"no valid validators found. skipping sending checkpoint")
+        else:
+            # create dendrite and transmit synapse
+            checkpoint_synapse = template.protocol.ValidatorCheckpoint(checkpoint=encoded_checkpoint)
+            validator_responses = await dendrite.forward(axons=validator_axons, synapse=checkpoint_synapse)
+
+            bt.logging.info(f"sending checkpoint from validator {self.wallet.hotkey.ss58_address}")
+
+            successes = 0
+            failures = 0
+            for response in validator_responses:
+                if response.successfully_processed:
+                    print(f"successfully processed ack from {response.validator_receive_hotkey}")
+                    successes += 1
+                else:
+                    failures += 1
+
+            bt.logging.info(f"{successes} responses succeeded")
+            bt.logging.info(f"{failures} responses failed")
+
+    def sync_positions_with_cooldown(self, auto_sync_enabled:bool, run_more:bool):
+        # Check if the time is right to sync signals
```

Comment:
Use the same time as existing autosync

If the time check fails, wipe out self.received_checkpoints_dict by setting it equal to {}. This takes care of the "reset" logic